content/doc/book/managing/plugins.adoc

```diff
@@ -4,13 +4,256 @@ layout: section
+logged-in Jenkins administrator can upload the file from within the web UI:
+
+. Navigate to the *Manage Jenkins* > *Manage Plugins* page in the web UI.
+. Click on the *Advanced* tab.
+. Choose the `.hpi` file under the *Upload Plugin* section.
+. *Upload* the plugin file.
+
+image::managing/plugin-manager-upload.png["Advanced tab in the Plugin Manager", role=center]
+
+Once a plugin file has been uploaded, the Jenkins master must be manually
+restarted in order for the changes to take effect.
+
+==== On the master
+
+Assuming a `.hpi` file has been downloaded from the Update Site footnoteref:[uc], a systems
+administrator who has access to the file system of the Jenkins master can also
+manually place the `.hpi` file in a specific location on the file system.
+
+Copy the downloaded `.hpi`` file into the `JENKINS_HOME/plugins` directory on
+the Jenkins master (for example, on Debian systems `JENKINS_HOME` is generally
+`/var/lib/jenkins`).
+
+The master will need to be restarted before the plugin is loaded and
+made available in the Jenkins environment.
+
 [NOTE]
 ====
-This is still very much a work in progress
+The names of the plugin directories in the Update Site footnoteref:[uc] are
+not always the same as the plugin's display name. Searching
+link:https://plugins.jenkins.io/[plugins.jenkins.io]
+for the desired plugin will provide the appropriate link to the `.hpi` files.
+====
+
+== Updating a plugin
+
+Updates are listed in the *Updates* tab of the *Manage Plugins* page and can be
+selected by checking the checkboxes of the desired plugin updates and clicking
+the *Download now and install after restart* button.
+
+image::managing/plugin-manager-update.png["Updates tab in the Plugin Manager", role=center]
+By default, the Jenkins master will check for updates from the Update Center
+once every 24 hours. To manually trigger a check for updates, simply click on
+the *Check now* button in the *Updates* tab.
+
+== Removing a plugin
+
+When a plugin is no longer used in a Jenkins environment, it is prudent to
+remove the plugin from the Jenkins master. This provides a number of benefits
+such as: reducing memory overhead at boot or runtime, reducing configuration
+options in the web UI, and removing the potential for future conflicts with new
+plugin updates.
+
+=== Uninstalling a plugin
+
+The simplest way to uninstall a plugin is to navigate to the *Installed* tab on
+the *Manage Plugins* page. From there, Jenkins will automatically determine
+which plugins are safe to uninstall, those which are not dependencies of other
+plugins, and present a button for doing so.
+
+image::managing/plugin-manager-uninstall.png["Installed tab in the Plugin Manager", role=center]
+
+
+A plugin may also be uninstalled by removing the corresponding `.hpi`
+file from the `JENKINS_HOME/plugins` directory on the master. The plugin will
+continue to function until the master has been restarted.
```

Comment:
This needs a WARNING block about how this can cause plugins that depend on a removed plugin to stop working.